### ISSUE TITLE
Sql Builder

### ISSUE THREAD

**Issue by brodotnet** [CollaboratingPlatypus/PetaPoco#653]
Hi Everyone, i want a ask about Sql builder.
I want to create database schema with postgreSql provider using sql builder function , but when this function executed with below code, the result always generated "0" as schema name 

When i try with this code `sql.Append("DECLARE myvar TEXT = @0 ; ","demo");`  the result it should be `DECLARE myvar TEXT = 'demo' `  instead it became like this `DECLARE myvar TEXT = '0` .
and i try with this code `sql.Append($"DECLARE myvar TEXT = '{schemaName}' ; ");` and its works.
but I want to avoid **sql injection**, therefore I use sql builder function.

**Fullcode below **

` Sql sql = new Sql();
  sql.Append("DO $$");
  sql.Append($"DECLARE myvar TEXT = '{name}' ; ");
  sql.Append("BEGIN EXECUTE 'CREATE SCHEMA ' || quote_ident(myvar); END $$;");
  await database.ExecuteAsync(sql);`
 
Thank you. 
  
![image](https://user-images.githubusercontent.com/73944368/185755726-4c9783b2-382d-4ef7-9539-fdead483bbb4.png)

![image](https://user-images.githubusercontent.com/73944368/185755806-e8c27b47-f70e-4b67-ac21-8ef7258cb55b.png)

**Comment by asherber** [comment]
I don't think this is an issue with PetaPoco -- the SQL fed to the command seems to be just what you asked for. Are you sure that Postgres allows the use of parameters in declaring a variable?
